go/cmd/dolt/commands/commit.go

```diff
@@ -376,38 +364,46 @@ func checkIsTerminal() bool {
 	return isTerminal
 }
 
-func buildInitalCommitMsg(ctx context.Context, dEnv *env.DoltEnv, suggestedMsg string) (string, error) {
+func buildInitalCommitMsg(ctx context.Context, sqlCtx *sql.Context, queryist cli.Queryist, suggestedMsg string) (string, error) {
 	initialNoColor := color.NoColor
 	color.NoColor = true
 
-	roots, err := dEnv.Roots(ctx)
+	schema, ri, err := queryist.Query(sqlCtx, "select table_name, status from dolt_status where staged == true")
 	if err != nil {
-		panic(err)
+		return "", err
 	}
-
-	stagedTblDiffs, notStagedTblDiffs, _ := diff.GetStagedUnstagedTableDeltas(ctx, roots)
-
-	ws, err := dEnv.WorkingSet(ctx)
+	stagedRows, err := sql.RowIterToRows(sqlCtx, schema, ri)
 	if err != nil {
 		return "", err
 	}
 
-	as, err := merge.GetMergeArtifactStatus(ctx, ws)
+	schema, ri, err = queryist.Query(sqlCtx, "select table_name, status from dolt_status where staged == false")
+	if err != nil {
+		return "", err
+	}
+	notStagedRows, err := sql.RowIterToRows(sqlCtx, schema, ri)
 	if err != nil {
-		return "", nil
+		return "", err
 	}
 
 	buf := bytes.NewBuffer([]byte{})
-	n := printStagedDiffs(buf, stagedTblDiffs, true)
-	n, err = PrintDiffsNotStaged(ctx, dEnv, buf, notStagedTblDiffs, true, false, n, as)
+	n := printStagedDiffs(buf, stagedRows, true)
+	n, err = PrintDiffsNotStaged(sqlCtx, queryist, buf, notStagedRows, true, false, n)
 	if err != nil {
 		return "", err
 	}
 
-	currBranch, err := dEnv.RepoStateReader().CWBHeadRef()
+	// get current branch
+	schema, ri, err = queryist.Query(sqlCtx, "select active_branch()")
```

Comment:
Seems like this would be useful elsewhere. Can you create a getCurrentBranch() function so other cli commands can get at it? Nick and Paval may already have one somewhere.